### PR TITLE
Update python version to 3.8 due to GitHub Actions update

### PR DESCRIPTION
**Pull request by yongtang**
This PR updates python version to 3.8.6 as GitHub Actions update (from 3.8.5)

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -315,10 +315,10 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v1
         with:
-          python-version: 3.8.5
+          python-version: 3.8.6
```
Comment (kvignesh1420) [inline .github/workflows/build.yml:318]: @yongtang why don't we leave it at 3.8?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -315,10 +315,10 @@ jobs:
...
       - name: Bazel on Windows
         env:
-          PYTHON_VERSION: 3.8.5
+          PYTHON_VERSION: 3.8.6
```
Comment (terrytangyuan) [inline .github/workflows/build.yml:321]: What's the context for this change?

**Comment by yongtang** [comment]
@terrytangyuan  @kvignesh1420 The error is caused by GitHub Actions recent change from 3.8.5 => 3.8.6, which caused the following (see logs in https://github.com/tensorflow/io/pull/1145/checks?check_run_id=1226724329) :
```
2020-10-08T15:33:41.4951411Z ##[error]Version 3.8.5 with arch x64 not found
Available versions:
2.7.18 (x86)
3.5.4 (x86)
3.6.8 (x86)
3.7.9 (x86)
3.8.6 (x86)
2.7.18 (x64)
3.5.4 (x64)
3.6.8 (x64)
3.7.9 (x64)
3.8.6 (x64)
```

**Comment by yongtang** [comment]
@terrytangyuan @kvignesh1420 On windows, we need `python3` command to run. However, GitHub Actions on Windows seems to only expose `python` command. As a result I use the following to copy `python` binary as `python3`:
https://github.com/tensorflow/io/blob/master/.github/workflows/build.yml#L330

**Comment by kvignesh1420** [comment]
@yongtang so was thinking if we can just leave it `3.8` and let Github actions take the latest version. What do you think?

**Comment by yongtang** [comment]
@kvignesh1420 We will also need to find python's absolute exe path:
https://github.com/tensorflow/io/blob/master/.github/workflows/build.yml#L330

It might be possible without explicitly calling out python's exact version?

**Comment by kvignesh1420** [comment]
@yongtang as per your ref: https://github.com/tensorflow/io/blob/master/.github/workflows/build.yml#L330

can we set the`PYTHON_VERSION` var to `python --version` before moving to python3?

**Comment by yongtang** [comment]
@kvignesh1420 updated. Please take a look.

**Review by terrytangyuan — APPROVED** [review]
LGTM now

**Review by kvignesh1420 — APPROVED** [review]
@yongtang  LGTM.

**Pull request merged**